### DISCUSSION TITLE
Projects to put ourselves in the context of

### DISCUSSION THREAD

**Discussion by bethac07** [bilayer-containers/bilayers#67] [Running lists of things]
We're not the only people out there building stuff! Which is not to say multiple solutions to the same problem aren't sometimes needed, but tracking similar-ish projects (ideally who we can learn from and/or collaborate).  

At least some of these are workflow runners, which we adamantly are not (and do not want to be), but in being workflow runners, they usually have some sort of a "tool description spec"

(When we go to write the preprint, eventually, future-us's will hopefully also be grateful)

**Comment by bethac07** [comment]
[Album](https://album.solutions/#catalogs) - [Preprint](https://arxiv.org/abs/2110.00601) includes Kyle Harrington, now of Napari (and frequent Eliceiri lab collab if I understand correctly). For imaging tools, which is neat! Love the idea of the catalog of tools/steps being federated.

**Comment by bethac07** [comment]
[EU Workflow Hub](https://workflowhub.eu/) - Repository of workflows. Lists about 30 [kinds of workflows](https://workflowhub.eu/workflow_classes) you can deposit (and counting)

**Comment by bethac07** [comment]
[Common Workflow Language](https://www.commonwl.org/index.html) - this was the thing I was most frequently asked how we intersect with. Certainly worth a discussion, before it's too late to pivot, if we should think about extending their [tool specification](https://www.commonwl.org/v1.2/CommandLineTool.html) rather than our own

**Comment by bethac07** [comment]
[Workflow Description Language](https://docs.openwdl.org/en/stable/) - it's made at Broad and is what Terra uses. Probably need a deeper dive on this sometime.

**Comment by bethac07** [comment]
[Dockstore](https://dockstore.org/) - "An App Store for Bioinformatics". Describes tools and supports description in "Common Workflow Language (CWL), Workflow Description Language (WDL), Nextflow (NFL), and Galaxy".

**Comment by bethac07** [comment]
biop-desktop [website](https://biop.github.io/biop-desktop-doc/) / [github of base image](https://github.com/BIOP/BIOP-desktop/blob/main/docker/Dockerfile-base) - one (ish) tool at a time, container launches a whole desktop (but seemingly differently than the one time Callum played with this, which was using xpra, they're using [TigerVNC](https://tigervnc.org/) and [noVNC](https://novnc.com/screenshots.html). Talked a bit about it with Romain at I2K but he said he was not the right person for deep tech stack questions (but could connect us to who was).

**Comment by bethac07** [comment]
Nextflow

**Comment by bethac07** [comment]
Snakemake

**Comment by bethac07** [comment]
Galaxy

**Comment by gnodar01** [comment]
[Many if not all the things](https://s.apache.org/existing-workflow-systems)

**Reply by bethac07** [reply]
"There are 362 standards"

**Comment by bethac07** [comment]
[Bio.tools](https://bio.tools/), a tool collection website

**Comment by bethac07** [comment]
https://github.com/Imaging-Server-Kit/imaging-server-kit
